contracts/pcv/utils/NamedStaticPCVDepositWrapper.sol

```diff
@@ -0,0 +1,202 @@
+    function _editDeposit(
+        uint256 index,
+        string calldata depositName,
+        uint256 usdAmount,
+        uint256 feiAmount,
+        uint256 underlyingTokenAmount,
+        address underlyingToken
+    ) internal {
+        DepositInfo storage updatePCVDeposit = pcvDeposits[index];
+
+        uint256 oldBalance = balance;
+        uint256 oldFEIBalance = feiReportBalance;
+
+        balance = balance - updatePCVDeposit.usdAmount + usdAmount;
+        feiReportBalance = feiReportBalance - updatePCVDeposit.feiAmount + feiAmount;
+
+        updatePCVDeposit.usdAmount = usdAmount + updatePCVDeposit.usdAmount;
+        updatePCVDeposit.depositName = depositName;
+        updatePCVDeposit.underlyingTokenAmount = underlyingTokenAmount;
+        updatePCVDeposit.underlyingToken = underlyingToken;
+
+        emit DepositEdited(index, depositName);
+        emit BalanceUpdate(oldBalance, balance, oldFEIBalance, feiReportBalance);
+    }
+
+    // ----------- Governor only state changing api -----------
+
+    /// @notice function to add a deposit
+    function addDeposit(
+        DepositInfo calldata newPCVDeposit
+    ) external onlyGovernorOrAdmin {
+        _addDeposit(newPCVDeposit);
+    }
+
+    /// @notice function to bulk add deposits
+    function bulkAddDeposits(
+        DepositInfo[] calldata newPCVDeposits
+    ) external onlyGovernorOrAdmin {
+        for (uint256 i = 0; i < newPCVDeposits.length; i++) {
+            _addDeposit(newPCVDeposits[i]);
+        }
+    }
+
+    /// @notice function to edit an existing deposit
+    function editDeposit(
+        uint256 index,
+        uint256 usdAmount,
+        uint256 feiAmount,
+        uint256 underlyingTokenAmount,
+        string calldata depositName,
+        address underlying
+    ) external onlyGovernorOrAdmin {
+        _editDeposit(
+            index,
+            depositName,
+            usdAmount,
+            feiAmount,
+            underlyingTokenAmount,
+            underlying
+        );
+    }
+
+    /// @notice function to remove a PCV Deposit
+    function removeDeposit(uint256 index) external onlyGovernorOrAdmin {
+        _removeDeposit(index);
+    }
+
+    // ----------- Getters -----------
+
+    /// @notice returns the current number of PCV deposits
+    function numDeposits() public view returns (uint256) {
+        return pcvDeposits.length;
+    }
+
+    /// @notice returns the resistant balance and FEI in the deposit
+    function resistantBalanceAndFei() public view override returns (uint256, uint256) {
+        return (balance, feiReportBalance);
+    }
+
+    /// @notice display the related token of the balance reported
+    function balanceReportedIn() public pure override returns (address) {
+        return Constants.USD;
+    }
+    
+    /// @notice function to return all of the different tokens deposited into this contract
+    function getAllUnderlying() public view returns (address[] memory) {
```

Comment:
This will return duplicates.
Suggest building an enumerableSet from the addresses and then using the method on enumerableSet to return all elements in the set - this will dedupe for you.

Reply:
This is pretty easy to implement, is this the right decision to make? My concerns are, you are trading off spending more on gas for something that can be filtered out client side.

I could add a view function `getAllUnderlyingUnique` that returns only unique underlying addresses and uses a double for loop to filter out duplicates.